bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -163,11 +164,11 @@ static inline void call_on_error(jobject callback, ucs_status_t status)
 void send_callback(void *request, ucs_status_t status)
 {
     struct jucx_context *ctx = (struct jucx_context *)request;
-    while(ctx->callback == NULL) {
+    while (ctx->callback == NULL || ctx->jucx_request == NULL) {
```

Comment:
( ) around == conditions
callback / jucx_request must be volatile
use ucs_memory_cpu_load_fence() [=read barrier] after while loop